api/workOrder/workOrderRouter.js

```diff
@@ -0,0 +1,494 @@
+ *        $ref: '#/components/responses/UnauthorizedError'
+ *      404:
+ *        description: 'WorkOrder not found'
+ *      200:
+ *        description: A workOrder object
+ *        content:
+ *          application/json:
+ *            schema:
+ *              type: object
+ *              properties:
+ *                message:
+ *                  type: string
+ *                  description: A message about the result
+ *                  example: workOrder created
+ *                workOrder:
+ *                  $ref: '#/components/schemas/WorkOrder'
+ */
+router.post('/', authRequired, async (req, res) => {
+  const workOrder = req.body;
+  if (workOrder) {
+    const id = workOrder.id || 0;
+    try {
+      await WorkOrders.findById(id).then(async (pf) => {
+        if (pf == undefined) {
+          //workOrder not found so lets insert it
+          await WorkOrders.create(workOrder).then((workOrder) =>
+            res
+              .status(200)
+              .json({ message: 'workOrder created', workOrder: workOrder[0] })
+          );
+        } else {
+          res.status(400).json({ message: 'workOrder already exists' });
+        }
+      });
+    } catch (e) {
+      console.error(e);
+      res.status(500).json({ message: e.message });
+    }
+  } else {
+    res.status(404).json({ message: 'WorkOrder missing' });
+  }
+});
+/**
+ * @swagger
+ * /workOrder:
+ *  put:
+ *    summary: Update a workOrder
+ *    security:
+ *      - okta: []
+ *    tags:
+ *      - workOrder
+ *    requestBody:
+ *      description: WorkOrder object to to be updated
+ *      content:
+ *        application/json:
+ *          schema:
+ *            $ref: '#/components/schemas/WorkOrder'
+ *    responses:
+ *      401:
+ *        $ref: '#/components/responses/UnauthorizedError'
+ *      404:
+ *        $ref: '#/components/responses/NotFound'
+ *      200:
+ *        description: A workOrder object
+ *        content:
+ *          application/json:
+ *            schema:
+ *              type: object
+ *              properties:
+ *                message:
+ *                  type: string
+ *                  description: A message about the result
+ *                  example: workOrder created
+ *                workOrder:
+ *                  $ref: '#/components/schemas/WorkOrder'
+ */
+router.put('/', authRequired, function (req, res) {
+  const workOrder = req.body;
+  if (workOrder) {
+    const id = workOrder.id || 0;
+    WorkOrders.findById(id)
+      .then(
+        WorkOrders.update(id, workOrder)
+          .then((updated) => {
+            res
+              .status(200)
+              .json({ message: 'workOrder created', workOrder: updated[0] });
+          })
+          .catch((err) => {
+            res.status(500).json({
+              message: `Could not update workOrder '${id}'`,
+              error: err.message,
+            });
+          })
+      )
+      .catch((err) => {
+        res.status(404).json({
```

Comment:
I don't believe this would be a 404 error, I believe it would be a 500 type of error as it's a server-side issue since the .catch will only be fired upon an error or rejected request (I believe)

Reply:
I will check this and get back to you.

Reply:
This is the catch to which I thought it would drop is the object hadn't been found (the preceding conversation). You are right that this should be a 500. Fixed.